tests/sympc/tensor/replicatedshare_tensor_test.py

```diff
@@ -136,3 +141,64 @@ def test_hook_property(get_clients) -> None:
 
     assert (rst.T.shares[0] == x.T).all()
     assert (rst.T.shares[1] == y.T).all()
+
+
+@pytest.mark.parametrize("parties", [2, 5, 11])
+def test_distribute_sharecount(get_clients, parties) -> None:
+    parties = get_clients(parties)
+    protocol = Falcon("semi-honest")
+    session = Session(protocol=protocol, parties=parties)
+    SessionManager.setup_mpc(session)
+
+    secret = 42.32
+
+    a = MPCTensor(secret=secret, session=session)
+
+    for share in a.share_ptrs:
+        assert len(share.get_shares().get()) == (len(parties) - 1)
+
+
+@pytest.mark.parametrize("parties", [3, 5, 11])
+@pytest.mark.parametrize("security", ["malicious", "semi-honest"])
+def test_rst_distribute_reconstruct(get_clients, parties, security) -> None:
+    parties = get_clients(parties)
+    protocol = Falcon(security)
+    session = Session(protocol=protocol, parties=parties)
+    SessionManager.setup_mpc(session)
+
+    secret = 42.32
+
+    a = MPCTensor(secret=secret, session=session)
+
+    assert np.allclose(secret, a.reconstruct(), atol=1e-5)
```

Comment:
Q: Since you did not run any operation, could you remove `atol`? (I think it might work also without it)

Reply:
```
E       assert 42.32 == tensor([42.3200])
E         +42.32
E         -tensor([42.3200])
```

It throws this error.

Reply:
It throws an error 

`>       assert secret==a.reconstruct().item()
E       assert 42.32 == 42.31999206542969
E         +42.32
E         -42.31999206542969`